### PR TITLE
[chore] Revise `TargetModel` & `TargetNamespaceModel`

### PR DESCRIPTION
**Pull request by gee-senbong**
## Description

Revise `TargetModel` & `TargetNamespaceModel` to make it closer to `FeatureModel` & `FeatureNamespaceModel`.

## Type of Change

Label this pull request to place it under the correct category in Release Notes:

|        **Pull Request Label**         | **Category in Release Notes** |
|:-------------------------------------:|:-----------------------------:|
|       `enhancement`, `feature`        |          🚀 Features          |
| `bug`, `refactoring`, `bugfix`, `fix` |    🔧 Fixes & Refactoring     |
|       `build`, `ci`, `testing`        |    📦 Build System & CI/CD    |
|              `breaking`               |      💥 Breaking Changes      |
|            `documentation`            |       📝 Documentation        |
|            `dependencies`             |    ⬆️ Dependencies updates    |

## Checklist

- [ ] I have read the [`CODE_OF_CONDUCT.md`](https://github.com/featurebyte/featurebyte/blob/main/CODE_OF_CONDUCT.md) and [`CONTRIBUTING.md`](https://github.com/featurebyte/featurebyte/blob/main/CONTRIBUTING.md) guides.
- [ ] I have written tests for the changes made.
- [ ] I have written docstrings in [NumpyDoc format](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)
- [ ] I have labeled my Pull Request correctly

### CODE REVIEW

**Review by jevonyeoh — APPROVED** [review]
sorry, think i just caused your a bunch of merge conflicts ><

File: featurebyte/models/target.py [hunk 1]
```diff
@@ -3,93 +3,57 @@
...
-    # Recipe
-    graph: Optional[QueryGraph] = Field(allow_mutation=False)
-    node_name: Optional[str] = Field(allow_mutation=False)
-
-    # These fields will either be inferred from the recipe, or manually provided by the user only if they're creating
-    # a target without a recipe.
-    horizon: Optional[str]
-    entity_ids: Optional[List[PydanticObjectId]] = Field(allow_mutation=False)
-
+    # list of IDs attached to this target
```
[the remark this answers is not in the document]
Comment (gee-senbong) [reply featurebyte/models/target.py:48]: This is used to split those ID related attributes (like `target_namespace_id`) vs non-ID related attributes. Will remove it if it is redundant to you.

[the remark this answers is not in the document]
Comment (jevonyeoh) [reply featurebyte/models/target.py:48]: oh gotcha i see what you mean. i read it expecting that there to be an attribute with a `List[PydanticObjectId]`, fine either way~

[the remark this answers is not in the document]
Comment (gee-senbong) [reply featurebyte/models/target.py:48]: Oh.. did not realize that it can be interpreted that way. Will revise the comment. 👍

**Pull request merged**